contracts/utils/Strings.sol

```diff
@@ -23,12 +25,41 @@ library Strings {
             temp /= 10;
         }
         bytes memory buffer = new bytes(digits);
-        uint256 index = digits;
-        temp = value;
+        while (value != 0) {
+            buffer[--digits] = bytes1(uint8(48 + uint256(value % 10)));
```

Comment:
I think this use of `--` is confusing for many people. I don't find it intuitive either.

```suggestion
            digits -= 1;
            buffer[digits] = bytes1(uint8(48 + uint256(value % 10)));
```